src/ocean/util/container/queue/FixedRingQueue.d

```diff
@@ -185,15 +187,16 @@ class FixedByteRingQueue : FixedRingQueueBase!(IByteQueue)
 
     ***************************************************************************/
 
-    bool push ( ubyte[] element )
+    bool push ( in ubyte[] element )
```

Comment:
I guess only question remaining from my side is why this takes `in ubyte[]` and not `in void[]` - you wouldn't need to do all those casts in tests otherwise.

Considering this is a breaking change targeting v4.x.x, maybe it is time to go void? :)

Reply:
I would like that, but it would be a rather intrusive change at a greater scope. It would in addition include
- `IByteQueue`
  - `ubyte[] push ( size_t size )`
  - `ubyte[] pop ( )`
  - `ubyte[] peek ( )`
- `NotifyingQueue`
  - `bool push ( size_t size, void delegate ( ubyte[] ) filler )`

It will most certainly break much more user code than just adding `in`. Right I'm torn here, `void[]` is much better IMHO.

Reply:
We could deprecate `IByteQueue` and all other `*Byte*Queue`s in favour of `*Data*Queue` which is `const`-correct and uses `void[]`. That way the `const`-correctnes would even got in v3.x.x.

Reply:
Sounds cool, though if we go this way, I would even leave `Byte*` part undeprecated first - it propagates up to swarm and forth which will annoy everyone again with dozens of unfixable deprecations.

Reply:
There is a complication with class naming. The classes in question are `FlexibleFileQueue`, `NotifyingQueue` and `QueueChain`. These classes depend on `IByteQueue` and need `IDataQueue` clones. What would be the best procedure here? I would like to avoid to add a useless `Data` infix to the class names.

Reply:
I guess I'm going to make it a breaking change in the next major release then.

Reply:
I am utterly irresponsible about naming things, ask @gavin-norman-sociomantic :trollface: 

As for breaking change - I would aim for addign new stuff in v3.x.x, deprecating old one at some point after 4.0.0 and removing completely in v5.x.x